nodegroups.tf

```diff
@@ -138,24 +121,36 @@ resource "aws_autoscaling_group" "eks" {
   target_group_arns    = var.target_group_arns
   health_check_type    = var.health_check_type
 
-  launch_template {
-    id      = aws_launch_template.eks_node_groups[each.key].id
-    version = "$Latest"
+  mixed_instances_policy {
+    dynamic "instances_distribution" {
+      for_each =  each.value.spot_nodes_enabled == true ? ["do it"] : []
```

Comment:
The `== true` isn't needed here, since `each.value.spot_nodes_enabled` is already a boolean type.